src/components/FileViewer/FileContextMenu.tsx

```diff
@@ -116,15 +118,36 @@ const FileContextMenu = forwardRef(
             }}
             disabled={file === null}
           />
-          <ContextMenuButton
-            className="w-full whitespace-nowrap rounded-none bg-gray-900 px-2.5 text-small font-medium"
-            label="Unzip"
-            onClick={() => {
-              unzipFile(file as ClientFile);
-              setShowContextMenu(false);
-            }}
-            disabled={file === null}
-          />
+          {((file === null || !(zipFileExtensions.includes(file.name.split('.').pop()!)) ) ? // if file name is null or file is not zip file
+            null : (<>
+            <ContextMenuButton
+              className="w-full whitespace-nowrap rounded-none bg-gray-900 px-2.5 text-small font-medium"
+              label="Unzip here"
+              onClick={() => {
+                unzipFile(file as ClientFile);
```

Comment:
you should pass this function another parameter of `UnzipOption` base on which option the user has selected